contrib/golang/filters/http/source/go/pkg/http/shim.go

```diff
@@ -208,6 +209,9 @@ func envoyGoFilterOnHttpDestroy(r *C.httpRequest, reason uint64) {
 	req := getRequest(r)
 	// do nothing even when req.panic is true, since filter is already destroying.
 	defer req.RecoverPanic()
+	if atomic.LoadInt32(&req.waitingOnEnvoy) != 0 {
```

Comment:
same as https://github.com/envoyproxy/envoy/pull/26883#discussion_r1174695680

Reply:
maybe we could use `CompareAndSwapInt32` before `sema.Done`, also need this protect in `envoyGoRequestSemaDec`.

Reply:
ok, thanks.